nwb_conversion_tools/datainterfaces/ecephys/blackrock/blackrockdatainterface.py

```diff
@@ -24,14 +22,7 @@ class BlackrockRecordingExtractorInterface(BaseRecordingExtractorInterface):
 
     @classmethod
     def get_source_schema(cls):
-        """Compile input schema for the RecordingExtractor."""
-        source_schema = get_schema_from_method_signature(
-            class_method=cls.__init__,
-            exclude=['block_index', 'seg_index']
-        )
-        source_schema['properties']['filename']['format'] = 'file'
-        source_schema['properties']['filename']['description'] = 'Path to Blackrock file.'
-        return source_schema
+        return get_schema_from_method_signature(cls.__init__)
```

Comment:
we could replace the upper part with:
```python
	source_schema = super().get_source_schema()
	source_schema['properties']['filename']['format'] = 'file'
    source_schema['properties']['filename']['description'] = 'Path to Blackrock file.'
    return source_schema
```
this addition is relevant for the GUI Forms.

Reply:
@luiztauffer Good catch.

How about this syntax:
```python
	source_schema = super().get_source_schema()
	source_schema['properties']['filename'].update(format='file', description='Path to Blackrock file.')
    return source_schema
```

We could also update `get_schema_from_method_signature` to include `format='file'` if the typehint includes the `Path` type.

Reply:
Folder modes can also be `Path` type, but you could automatically determine the difference based on if the suffix is empty. _I.e._, `Path("./Test").suffix` returns `""`

Reply:
we couldn't really parse it because this is a definition, comes prior to the user inputs. In fact, the user inputs should be validated by that definition (we're not validating this right now)